src/kernel/tty.zig

```diff
@@ -451,1794 +56,28 @@ pub fn print(comptime format: []const u8, args: var) void {
 }
 
 ///
-/// Move up a page. This will copy the page above to the video buffer. Will keep trace of which
-/// page is being displayed.
-///
-pub fn pageUp() void {
-    if (page_index < TOTAL_NUM_PAGES - 1) {
-        // Copy page to display
-        page_index += 1;
-        // Bounds have been checked, so shouldn't error
-        videoCopy(START_OF_DISPLAYABLE_REGION, pages[page_index][0..TOTAL_CHAR_ON_PAGE], TOTAL_CHAR_ON_PAGE) catch |e| {
-            log.logError("TTY: Error moving page up. Error: {}\n", .{e});
-        };
-        displayPageNumber();
-        vga.disableCursor();
-    }
-}
-
-///
-/// Move down a page. This will copy the page bellow to the video buffer. Will keep trace of which
-/// page is being displayed.
-///
-pub fn pageDown() void {
-    if (page_index > 0) {
-        // Copy page to display
-        page_index -= 1;
-        // Bounds have been checked, so shouldn't error
-        videoCopy(START_OF_DISPLAYABLE_REGION, pages[page_index][0..TOTAL_CHAR_ON_PAGE], TOTAL_CHAR_ON_PAGE) catch |e| {
-            log.logError("TTY: Error moving page down. Error: {}\n", .{e});
-        };
-
-        displayPageNumber();
-        if (page_index == 0) {
-            vga.enableCursor();
-            updateCursor();
-        } else {
-            vga.disableCursor();
-        }
-    }
-}
-
-///
-/// This clears the entire screen with blanks using the current colour. This will also save the
-/// screen to the pages so can scroll back down.
-///
-pub fn clearScreen() void {
-    // Move all the rows up
-    // This is within bounds, so shouldn't error
-    pagesMoveRowsUp(ROW_TOTAL) catch |e| {
-        log.logError("TTY: Error moving all pages up. Error: {}\n", .{e});
-    };
-
-    // Clear the screen
-    var i: u16 = START_OF_DISPLAYABLE_REGION;
-    while (i < VIDEO_BUFFER_SIZE) : (i += 1) {
-        video_buffer[i] = blank;
-    }
-
-    // Set the cursor to below the logo
-    column = 0;
-    row = ROW_MIN;
-    updateCursor();
-}
-
-///
-/// This moves the software and hardware cursor to the left by one.
-///
-pub fn moveCursorLeft() void {
-    if (column == 0) {
-        if (row != 0) {
-            column = vga.WIDTH - 1;
-            row -= 1;
-        }
-    } else {
-        column -= 1;
-    }
-
-    updateCursor();
-}
-
-///
-/// This moves the software and hardware cursor to the right by one.
+/// Clear the screen by printing a space at each cursor position. Sets the cursor to the top left (0, 0)
 ///
-pub fn moveCursorRight() void {
-    if (column == (vga.WIDTH - 1)) {
-        if (row != (vga.HEIGHT - 1)) {
-            column = 0;
-            row += 1;
+pub fn clear() void {
+    var row: u8 = 0;
+    while (row < tty.rows) : (row += 1) {
+        var col: u8 = 0;
+        while (col < tty.cols) : (col += 1) {
+            print(" ", .{});
         }
-    } else {
-        column += 1;
+        print("\n", .{});
     }
-
-    updateCursor();
+    tty.setCursor(0, 0);
```

Comment:
Oh yes I didn't think to look for one. I'll add an optional clear function to the TTY struct so that archs without a specialised function can use the default implementation.